### ISSUE TITLE
Unexpected behaviour with removeMode data or run

### ISSUE THREAD

**Issue by wrosenuance** [fabric8io/docker-maven-plugin#1214]
### removeMode = data
The documentation and code in the docker:remove mojo suggests that images without a run configuration are considered data images, and can be removed when removeMode = data. However, the code seems to work against that.

Starting with RemoveMojo.java:
```java
    private boolean imageShouldBeRemoved(ImageConfiguration image) {
    // ... snipped ...
        if ("data".equalsIgnoreCase(removeMode)) {
            return image.isDataImage();
        }
```

Then in ImageConfiguration.java:

```java
    public boolean isDataImage() {
        // If there is no explicit run configuration, its a data image
        // TODO: Probably add an explicit property so that a user can indicated whether it
        // is a data image or not on its own.
        return getRunConfiguration() == null;
    }
```

But, looking at getRunConfiguration():

```java
    public RunImageConfiguration getRunConfiguration() {
        return (run == null) ? RunImageConfiguration.DEFAULT : run;
    }
```

It is never possible for getRunConfiguration() to return non-null, therefore never possible for isDataImage() to return true, therefore never possible for removeMode=data to ever remove an image.

Probably the fix is to have isDataImage check the field directly (i.e. run == null vs getRunConfiguration() == null), but this code has been in place 4-5 years, so I'm wondering if this is useful functionality to anyone, and/or actually working correctly somehow.

### removeMode = run
The description for removeMode = run is that it removes images with a run configuration. Yet the code in RemoveMojo.java is:

```java
    private boolean imageShouldBeRemoved(ImageConfiguration image) {
    // ... snipped ...
        if ("run".equalsIgnoreCase(removeMode)) {
            return image.getRegistry() != null;
        }
```

The presence/absence of a registry seems like a strange test for "is this container runnable", as the local docker server need not be set as the registry. It also doesn't guarantee there is a run configuration. But that said, the defaulting of the run configuration as above means that there doesn't need to be a run configuration in the POM for an image to be runnable, so this raises the question of whether a run image is actually one without a build configuration, rather than one with a run configuration.

Overall though, I'm not sure what to make of these two options. Do they make sense to keep?

* d-m-p version : 0.29-SNAPSHOT

**Comment by rhuss** [comment]
You are right, these both options are from the very early days of d-m-p where the idea of a pure 'data' image where introduced. This was one idea which I used for implementing the "Configuration Container" pattern (see e.g. https://github.com/ro14nd-talks/docker-patterns/blob/master/docker-patterns.pdf, also used to some degree in http://shop.oreilly.com/product/0636920242598.do), but I think it never really took off.

I suggest to fix it for now (both as you suggested, e.g. fix for "data", check for no build config for a "run" image) but then deprecate it and remove the notion of a "data" image in a later sweep.
